src/engraving/io/msczreader.cpp

```diff
@@ -129,32 +181,73 @@ const MsczReader::Meta& MsczReader::meta() const
         return m_meta;
     }
 
-    QVector<MQZipReader::FileInfo> files = reader()->fileInfoList();
-    for (const MQZipReader::FileInfo& fi : files) {
-        if (fi.isFile) {
-            if (fi.filePath.endsWith(".mscx")) {
-                m_meta.mscxFileName = fi.filePath;
-            } else if (fi.filePath.startsWith("Pictures/")) {
-                m_meta.imageFilePaths.push_back(fi.filePath);
+    auto fileList = [this]() {
+        QStringList files;
+        switch (m_mode) {
+        case Mode::Zip: {
+            QVector<MQZipReader::FileInfo> fileInfoList = reader()->fileInfoList();
+
+            if (reader()->status() != MQZipReader::NoError) {
+                LOGE() << "failed read meta, status: " << reader()->status();
+            }
+
+            for (const MQZipReader::FileInfo& fi : fileInfoList) {
+                if (fi.isFile) {
+                    files << fi.filePath;
+                }
             }
+        } break;
+        case Mode::Dir: {
+            QString root = rootPath();
+            QDirIterator::IteratorFlags flags = QDirIterator::Subdirectories;

```

Comment:
isn't it making sense to use framework::filesystem for that mode?

Reply:
At the moment, the engraving module does not depend on anything and for now it is better to keep it that way.